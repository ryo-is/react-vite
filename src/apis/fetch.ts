import type { Prefecture } from '../stores/jotai/population/types';

type GetPrefecturesResponse = {
  result: Prefecture[];
  statusCode?: number;
  message: string | null;
};
export const getPrefectures = async (): Promise<GetPrefecturesResponse> => {
  const res = await fetch(
    'https://opendata.resas-portal.go.jp/api/v1/prefectures',
    {
      headers: {
        'X-API-KEY': import.meta.env.VITE_API_KEY as string,
      },
    },
  );
  const resJson = (await res.json()) as GetPrefecturesResponse;
  return resJson;
};

type GetPopulationsResponse = {
  result: {
    boundaryYear: number;
    data: (
      | {
          label: string;
          data: {
            year: number;
            value: number;
          }[];
        }
      | {
          label: string;
          data: {
            year: number;
            value: number;
            rate: number;
          }[];
        }
    )[];
  };
  statusCode?: number;
  message: string | null;
};

export const getPopulations = async (
  prefectures: Prefecture[],
): Promise<GetPopulationsResponse[]> => {
  const promises = prefectures.map((pref) =>
    fetch(
      `https://opendata.resas-portal.go.jp/api/v1/population/composition/perYear?cityCode=-&prefCode=${pref.prefCode}`,
      {
        headers: {
          'X-API-KEY': import.meta.env.VITE_API_KEY as string,
        },
      },
    ),
  );
  const results = await Promise.all(promises);
  const toJsonPromises = results.map((r) => r.json());
  const populationJsons =
    await Promise.all<GetPopulationsResponse>(toJsonPromises);
  return populationJsons;
};
