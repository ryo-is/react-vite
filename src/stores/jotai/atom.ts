import { atom } from 'jotai';
import { atomWithStorage } from 'jotai/utils';

export const countAtom = atom({ count: 0 });
export const getCountValue = atom((get) => get(countAtom).count);
export const getDoubleCountValue = atom((get) => get(countAtom).count * 2);

export const countAtomWithStorageAtom = atomWithStorage(
  'countAtomWithStorage',
  0
);
