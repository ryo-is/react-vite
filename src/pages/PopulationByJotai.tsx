/* eslint-disable react-hooks/exhaustive-deps */
import { useAtom, useSetAtom } from 'jotai';
import { useEffect } from 'react';
import {
  CartesianGrid,
  Legend,
  Line,
  LineChart,
  ResponsiveContainer,
  Tooltip,
  XAxis,
  YAxis,
} from 'recharts';

import {
  useClearSelectedPrefectures,
  usePopulations,
  usePopulationSeries,
  usePrefectures,
  useSelectedPrefectures,
} from '../stores/jotai/population/atom';
import { Prefecture } from '../stores/jotai/population/types';
import { useToast } from '../stores/jotai/toast/atom';

const lineColors = [
  '#b33dc6',
  '#27aeef',
  '#87bc45',
  '#bdcf32',
  '#ede15b',
  '#edbf33',
  '#ef9b20',
  '#f46a9b',
  '#ea5545',
];
const getStrokeColor = (index: number) =>
  lineColors[(index + 1) % lineColors.length];

export const PopulationByJotai = () => {
  const [prefectures, fetchPrefectures] = useAtom(usePrefectures);
  const [selectedPrefectures, setSelectedPrefectures] = useAtom(
    useSelectedPrefectures,
  );
  const fetchPopulations = useSetAtom(usePopulations);
  const [series] = useAtom(usePopulationSeries);
  const clearSelectedPrefectures = useSetAtom(useClearSelectedPrefectures);
  const setToast = useSetAtom(useToast);

  const isChecked = (code: number) => {
    const idx = selectedPrefectures.findIndex((p) => p.prefCode === code);
    return idx >= 0;
  };

  const handleOnChange = (checked: boolean, pref: Prefecture) => {
    setSelectedPrefectures({ checked, pref });
  };

  const handleOnClear = () => {
    clearSelectedPrefectures();
    setToast({ message: 'done clear all', status: 'info' });
  };

  useEffect(() => {
    void (async () => {
      await fetchPrefectures();
    })();
  }, []);

  useEffect(() => {
    void (async () => {
      await fetchPopulations();
    })();
  }, [selectedPrefectures]);

  return (
    <div className="flex space-x-4">
      <div className="w-1/5">
        <div className="mb-4 flex items-center justify-between">
          <div>Prefectures</div>
          <button
            type="button"
            className="btn btn-info"
            onClick={() => handleOnClear()}
          >
            clear all
          </button>
        </div>
        <div className="flex max-h-[80vh] w-full flex-col overflow-scroll rounded-lg border-2 border-zinc-500 p-4">
          {prefectures.length > 0 ? (
            <>
              {prefectures.map((p) => (
                <div
                  key={p.prefCode}
                  className="my-1 flex items-center space-x-2"
                >
                  <input
                    type="checkbox"
                    id={p.prefName}
                    className="hidden"
                    checked={isChecked(p.prefCode)}
                    onChange={(e) => handleOnChange(e.target.checked, p)}
                  />
                  <label
                    className={`${
                      isChecked(p.prefCode)
                        ? 'border-zinc-400 bg-zinc-300 text-zinc-800'
                        : 'border-zinc-500'
                    } label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0`}
                    htmlFor={p.prefName}
                  >
                    {p.prefName}
                  </label>
                </div>
              ))}
            </>
          ) : (
            <div>Loading...</div>
          )}
        </div>
      </div>
      <div className="w-4/5">
        {series.length > 0 ? (
          <ResponsiveContainer
            width="100%"
            height="100%"
            className="text-zinc-700"
          >
            <LineChart
              width={500}
              height={300}
              margin={{
                top: 5,
                right: 30,
                left: 50,
                bottom: 5,
              }}
            >
              <CartesianGrid strokeDasharray="3 3" opacity={0.5} />
              <XAxis
                dataKey="year"
                type="category"
                allowDuplicatedCategory={false}
                stroke="#e4e4e7"
              />
              <YAxis stroke="#e4e4e7" />
              <Tooltip />
              <Legend />
              {series.map((s, i) => (
                <Line
                  key={s.label}
                  type="monotone"
                  data={s.data}
                  name={s.label}
                  dataKey="value"
                  strokeWidth={4}
                  stroke={getStrokeColor(i)}
                  dot={{ r: 4, fill: getStrokeColor(i) }}
                  activeDot={{ r: 8 }}
                  animationDuration={500}
                  animationEasing="ease-in-out"
                />
              ))}
            </LineChart>
          </ResponsiveContainer>
        ) : (
          <div className="p-4">Please select prefecture.</div>
        )}
      </div>
    </div>
  );
};
