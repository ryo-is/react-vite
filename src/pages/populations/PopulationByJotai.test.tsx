import { cleanup, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'jotai';
import { useHydrateAtoms } from 'jotai/utils';
import type { ReactNode } from 'react';
import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';

import { PopulationByJotai } from './PopulationByJotai';
import * as Fetch from '../../apis/fetch';
import {
  mockGetPopulationsResponse,
  mockGetPrefecturesResponse,
} from '../../helper/mockResponses';
import {
  populationAtom,
  prefectureAtom,
  selectedPrefectureAtom,
} from '../../stores/jotai/population/atom';
import type {
  Population,
  Prefecture,
} from '../../stores/jotai/population/types';

const user = userEvent.setup();

type ProviderProps = {
  initialValues: Array<
    | [typeof prefectureAtom, Prefecture[]]
    | [typeof populationAtom, Population[]]
  >;
  children: ReactNode;
};

const HydrateAtoms = ({ initialValues, children }: ProviderProps) => {
  useHydrateAtoms(initialValues);
  return children;
};

const TestProvider = ({ initialValues, children }: ProviderProps) => (
  <Provider>
    <HydrateAtoms initialValues={initialValues}>{children}</HydrateAtoms>
  </Provider>
);

const initialValues: ProviderProps['initialValues'] = [
  [prefectureAtom, []],
  [selectedPrefectureAtom, []],
  [populationAtom, []],
];

const ToastProvider = () => (
  <TestProvider initialValues={initialValues}>
    <PopulationByJotai />
  </TestProvider>
);

describe('PopulationByJotai', () => {
  beforeEach(() => {
    vi.mock('recharts');
    vi.spyOn(Fetch, 'getPrefectures').mockResolvedValue(
      mockGetPrefecturesResponse,
    );
    vi.spyOn(Fetch, 'getPopulations').mockResolvedValue([
      mockGetPopulationsResponse,
    ]);
  });

  afterEach(() => {
    vi.clearAllMocks();
    cleanup();
  });

  test('should match to the snapshot', async () => {
    const { asFragment } = await waitFor(() => render(<ToastProvider />));
    expect(asFragment()).toMatchSnapshot();
  });

  test('should call getPrefectures when rendering', async () => {
    await waitFor(() => render(<ToastProvider />));

    expect(Fetch.getPrefectures).toHaveBeenCalled();
  });

  test('should get population when select prefecture', async () => {
    await waitFor(() => render(<ToastProvider />));
    screen.debug();
    const cb = screen.getByRole('checkbox', { name: '北海道' });
    await user.click(cb);

    expect(Fetch.getPopulations).toHaveBeenCalledWith([
      {
        prefCode: 1,
        prefName: '北海道',
      },
    ]);
  });
});
