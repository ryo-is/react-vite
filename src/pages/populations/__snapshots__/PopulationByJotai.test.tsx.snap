// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PopulationByJotai > should match to the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex space-x-4"
  >
    <div
      class="w-1/5"
    >
      <div
        class="mb-4 flex items-center justify-between"
      >
        <div>
          Prefectures
        </div>
        <button
          class="btn btn-info"
          type="button"
        >
          clear all
        </button>
      </div>
      <div
        class="flex max-h-[80vh] w-full flex-col overflow-scroll rounded-lg border-2 border-zinc-500 p-4"
      >
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="北海道"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="北海道"
          >
            北海道
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="青森県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="青森県"
          >
            青森県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="岩手県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="岩手県"
          >
            岩手県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="宮城県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="宮城県"
          >
            宮城県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="秋田県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="秋田県"
          >
            秋田県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="山形県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="山形県"
          >
            山形県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="福島県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="福島県"
          >
            福島県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="茨城県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="茨城県"
          >
            茨城県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="栃木県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="栃木県"
          >
            栃木県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="群馬県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="群馬県"
          >
            群馬県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="埼玉県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="埼玉県"
          >
            埼玉県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="千葉県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="千葉県"
          >
            千葉県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="東京都"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="東京都"
          >
            東京都
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="神奈川県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="神奈川県"
          >
            神奈川県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="新潟県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="新潟県"
          >
            新潟県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="富山県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="富山県"
          >
            富山県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="石川県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="石川県"
          >
            石川県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="福井県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="福井県"
          >
            福井県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="山梨県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="山梨県"
          >
            山梨県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="長野県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="長野県"
          >
            長野県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="岐阜県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="岐阜県"
          >
            岐阜県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="静岡県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="静岡県"
          >
            静岡県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="愛知県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="愛知県"
          >
            愛知県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="三重県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="三重県"
          >
            三重県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="滋賀県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="滋賀県"
          >
            滋賀県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="京都府"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="京都府"
          >
            京都府
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="大阪府"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="大阪府"
          >
            大阪府
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="兵庫県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="兵庫県"
          >
            兵庫県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="奈良県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="奈良県"
          >
            奈良県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="和歌山県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="和歌山県"
          >
            和歌山県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="鳥取県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="鳥取県"
          >
            鳥取県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="島根県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="島根県"
          >
            島根県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="岡山県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="岡山県"
          >
            岡山県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="広島県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="広島県"
          >
            広島県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="山口県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="山口県"
          >
            山口県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="徳島県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="徳島県"
          >
            徳島県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="香川県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="香川県"
          >
            香川県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="愛媛県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="愛媛県"
          >
            愛媛県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="高知県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="高知県"
          >
            高知県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="福岡県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="福岡県"
          >
            福岡県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="佐賀県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="佐賀県"
          >
            佐賀県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="長崎県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="長崎県"
          >
            長崎県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="熊本県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="熊本県"
          >
            熊本県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="大分県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="大分県"
          >
            大分県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="宮崎県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="宮崎県"
          >
            宮崎県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="鹿児島県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="鹿児島県"
          >
            鹿児島県
          </label>
        </div>
        <div
          class="my-1 flex items-center space-x-2"
        >
          <input
            class="hidden"
            id="沖縄県"
            type="checkbox"
          />
          <label
            class="border-zinc-500 label-text w-full cursor-pointer  rounded-md border-2 p-4 first:mt-0"
            for="沖縄県"
          >
            沖縄県
          </label>
        </div>
      </div>
    </div>
    <div
      class="w-4/5"
    >
      <div
        class="p-4"
      >
        Please select prefecture.
      </div>
    </div>
  </div>
</DocumentFragment>
`;
